addons/website_forum/static/src/js/website_forum.js

```diff
@@ -29,7 +30,7 @@ publicWidget.registry.websiteForum = publicWidget.Widget.extend({
         'click .o_wforum_validation_queue a[href*="/validate"]': '_onValidationQueueClick',
         'click .o_wforum_validate_toggler:not(.karma_required)': '_onAcceptAnswerClick',
         'click .o_wforum_favourite_toggle': '_onFavoriteQuestionClick',
-        'click .comment_delete': '_onDeleteCommentClick',
+        'click .comment_delete:not(.karma_required)': '_onDeleteCommentClick',
```

Comment:
Hi @amdi-odoo , I was wondering about this change, then tried to put a large karma minimum to delete comments : https://tinyurl.com/24lvyhqk
Some weird stuff happened. (Maybe a right [discrepancy](https://tinyurl.com/24lvyhqk))

Was I supposed to see some karma_required class on that button? Hence your change? Cheers!

Reply:
Hello @Odoonan :slightly_smiling_face: 

Yes there was supposed to have a `karma_required` class on the button here :smiling_face_with_tear: 
[Here](https://github.com/odoo/odoo/blob/master/addons/website_forum/static/src/js/website_forum.js#L18) in the js, we show a warning as soon as the user clicks on an element with the `karma_required` class. 
That's why I added the `:not(.karma_required)` to prevent opening the dialog if the warning will be shown. (`karma required` class on the button template [here](https://github.com/odoo/odoo/blob/master/addons/website_forum/views/forum_forum_templates_tools.xml#L20) )

That's a mess but I think I got what was wrong, there is a mismatch between the xml karma condition [[1]](https://github.com/odoo/odoo/blob/master/addons/website_forum/views/forum_forum_templates_post.xml#L579) [[2]](https://github.com/odoo/odoo/blob/master/addons/website_forum/views/forum_forum_templates_tools.xml#L20) and the [python](https://github.com/odoo/odoo/blob/master/addons/website_forum/models/forum_post.py#L714) one. 
So when you clicked on delete, the xml one passed but the exception was raised in the python.

The xml condition is using the `can_unlink` and `karma_unlink` fields but that's actually fields related to the `post` not the `comment`... So currently a non-admin user that has enough karma to delete a comment have a warning shown (showing the **_post_** deletion required karma) even though he can perform the action... https://tinyurl.com/24ldzwhf 

=> I'm going to create a new task to fix the xml condition in stable as this is the case since 15.0.
We could continue with this one knowing the karma xml variable will be fixed (the dialog won't be shown when the warning is displayed) or we could wait for the fix to be merged :slightly_smiling_face: your choice!

Thanks for noticing! Cheers!